public/app/core/components/AppChrome/menuPreference.ts

```diff
@@ -0,0 +1,24 @@
+import { store } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+export const DOCKED_LOCAL_STORAGE_KEY = 'grafana.navigation.docked';
+
+export enum MenuDockState {
+  Undocked,
+  Docked,
+  AutoDocked,
+}
+
+export function getMegaMenuDockedState() {
+  if (window.innerWidth < config.theme2.breakpoints.values.xl) {
+    return MenuDockState.Undocked;
+  }
+
+  const preference = store.getBool(DOCKED_LOCAL_STORAGE_KEY, null);
```

Comment:
ah now i see why 😄 wonder whether we should just use `store.get` instead to see if there's a key set? 🤔 
e.g. something like
```tsx
const hasPreference = store.get(DOCKED_LOCAL_STORAGE_KEY) !== undefined

if (!hasPreference) {
  return window.innerWidth >= config.theme2.breakpoints.values.xxl ? MenuDockState.AutoDocked : MenuDockState.Undocked;
}

const preference = store.getBool(DOCKED_LOCAL_STORAGE_KEY, Boolean(window.innerWidth >= config.theme2.breakpoints.values.xxl)

return preference ? MenuDockState.Docked : MenuDockState.Undocked;
```

Reply:
Yeah - that's probably better.